### PR TITLE
feat(azure): Add support for Azure Block Blob Storage Account

### PR DESCRIPTION
**Pull request by ekifel**
Issue #574 

Output examples: in golden test. 

Details: 

- Terraform supports Premium and Standard account tiers.
- For Standard tier possible redundancy is: LRS, GRS, RA_GRS
- For Premium tier possible redundancy is: LRS, ZRS

### CODE REVIEW

File: internal/providers/terraform/azure/storage_account.go [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+	if u != nil && u.Get("storage_gb").Exists() {
+		capacity = decimalPtr(decimal.NewFromInt(u.Get("storage_gb").Int()))
+
+		if accessTier == "Hot" {
+			dataStorageTiers := []int{51200, 512000}
+			dataStorageQuantities := usage.CalculateTierBuckets(*capacity, dataStorageTiers)
+
+			costComponents = append(costComponents, blobDataStorageCostComponent(location, "Capacity (first 50TB)", skuName, "0", productName, dataMeterName, &dataStorageQuantities[0]))
+			if dataStorageQuantities[1].GreaterThan(decimal.Zero) {
```
[the remark this answers is not in the document]
Comment (ekifel) [reply internal/providers/terraform/azure/storage_account.go:86]: Nope, coz CalculateTierBuckets always returns correct number of array elements (zeros, if tier is not using).

File: internal/providers/terraform/azure/storage_account.go [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+	accountTier := d.Get("account_tier").String()
+	accessTier := "Hot"
+	if d.Get("access_tier").Type != gjson.Null {
+		accessTier = d.Get("access_tier").String()
+	}
+
+	productName := map[string]string{
+		"Standard": "Blob Storage",
+		"Premium":  "Premium Block Blob",
+	}[accountTier]
```
[the remark this answers is not in the document]
Comment (ekifel) [reply internal/providers/terraform/azure/storage_account.go:45]: In this case, TF will show user an error: `Error: expected account_tier to be one of [Standard Premium]`

**Comment by ekifel** [comment]
@aliscott, this is ready for review

File: internal/providers/terraform/azure/storage_account.go [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+	if productName == "" {
+		log.Warnf("Unrecognized account tier for resource %s: %s", d.Address, accountTier)
+		return nil
+	}
+
+	validPremiumReplicationTypes := []string{"ZRS", "LRS"}
+	validStandardReplicationTypes := []string{"LRS", "GRS", "RAGRS"}
+
+	if accessTier == "Premium" {
+		if !Contains(validPremiumReplicationTypes, accountReplicationType) || !Contains(validStandardReplicationTypes, accountReplicationType) {
```
Comment (aliscott) [inline internal/providers/terraform/azure/storage_account.go:55]:
```suggestion
	if accessTier == "Premium" && (!Contains(validPremiumReplicationTypes, accountReplicationType) || !Contains(validStandardReplicationTypes, accountReplicationType)) {
```

File: internal/providers/terraform/azure/storage_account.go [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+		var unknown *decimal.Decimal
+
+		costComponents = append(costComponents, blobDataStorageCostComponent(location, "Capacity", skuName, "0", productName, unknown))
+	}
+
+	if u != nil && u.Get("monthly_write_operations").Exists() {
+		writeOperations = decimalPtr(decimal.NewFromInt(u.Get("monthly_write_operations").Int()))
+	}
+	meterName := "/Write Operations$/"
+	costComponents = append(costComponents, blobOperationsCostComponent(location, "Write operations", "10K operations", skuName, meterName, productName, writeOperations, 10000))
```
Comment (aliscott) [inline internal/providers/terraform/azure/storage_account.go:98]:
```suggestion
	costComponents = append(costComponents, blobOperationsCostComponent(
		location,
		"Write operations",
		"10K operations",
		skuName,
		"/Write Operations$/",
		productName,
		writeOperations,
		10000,
	))
```
This doesn't need to change. I think breaking this into lines makes it easier to read and doesn't need a redundant `meterName` var. But it doesn't affect the code, it's just a personal stylistic thing, so will leave it up to you.

**Review by aliscott — APPROVED** [review]
This looks great, thanks @ekifel ! A couple of stylistic comments, but nothing that really needs changed.

File: internal/providers/terraform/azure/storage_account.go (hunk 4 above)
Comment (ekifel) [reply internal/providers/terraform/azure/storage_account.go]: Yes, u are right - it looks better :) 
+ I noticed and fixed one nuance here, I don't know why, but Hot RA-GRS List and Create Container Operations component does not exists

**Comment by aliscott** [comment]
Thanks @ekifel!

**Pull request merged**